server/src/api/user.rs

```diff
@@ -905,3 +972,150 @@ impl Perform<LoginResponse> for Oper<PasswordChange> {
     })
   }
 }
+
+impl Perform<PrivateMessageResponse> for Oper<CreatePrivateMessage> {
+  fn perform(&self, conn: &PgConnection) -> Result<PrivateMessageResponse, Error> {
+    let data: &CreatePrivateMessage = &self.data;
+
+    let claims = match Claims::decode(&data.auth) {
+      Ok(claims) => claims.claims,
+      Err(_e) => return Err(APIError::err(&self.op, "not_logged_in").into()),
+    };
+
+    let user_id = claims.id;
+
+    let hostname = &format!("https://{}", Settings::get().hostname);
+
+    // Check for a site ban
+    if UserView::read(&conn, user_id)?.banned {
+      return Err(APIError::err(&self.op, "site_ban").into());
+    }
```

Comment:
Do you do this in every API call? If the user is banned, we should just force logout and then deny the login.

Reply:
Not every call, just all the creation related ones. I still want to enable the login, since there's the possiblity for an unban.

Reply:
If the user is unbanned, they can login again. Much simpler, and I think this is how most sites do it.

Reply:
I'd still want them to be able to see their comments and stuff tho, like if reddit bans someone I think it might still let you log in, but it has a disclaimer saying you are. Its just a different way to do it.